src/sage/crypto/key_exchange/diffie_hellman.py

```diff
@@ -0,0 +1,314 @@
+r"""
+Diffie-Hellman Public Key Exchange Scheme
+
+This module contains a toy implementation of the Diffie-Hellman public key exchange
+scheme.
+
+AUTHORS:
+
+- Vincent Macri (2024-07-15): initial version
+"""
+# ****************************************************************************
+#       Copyright (C) 2024 Vincent Macri <vincent.macri@ucalgary.ca>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#                  https://www.gnu.org/licenses/
+# ****************************************************************************
+
+from sage.misc.superseded import experimental
+
+from sage.crypto.key_exchange.key_exchange import KeyExchangeScheme
+
+from sage.arith.misc import is_prime
+from sage.misc.prandom import randint
+from sage.rings.integer import Integer
+from sage.rings.finite_rings.finite_field_constructor import GF
+from sage.rings.finite_rings.finite_field_prime_modn import FiniteField_prime_modn
+from sage.rings.finite_rings.integer_mod import IntegerMod_int
+from sage.structure.proof.proof import WithProof
+
+from typing import Union
+
+class DiffieHellman(KeyExchangeScheme):
+
+    @experimental(37305)
+    def __init__(self, p: Integer, g: Union[Integer, IntegerMod_int], proof: bool = True):
```

Comment:
80 cols format & default values for function parameters should stick to the `=`. We should check if the style says to indent with 4 spaces or to align with the first parameter.
```suggestion
    def __init__(self, p: Integer, g: Union[Integer, IntegerMod_int], proof:
        bool=True):
```